awswrangler/athena/_read.py

```diff
@@ -835,6 +848,10 @@ def read_sql_query(
         max_remote_cache_entries=max_remote_cache_entries,
     )
     _logger.debug("cache_info:\n%s", cache_info)
+
+    if cache_info.has_valid_cache is True and client_request_token:
+        raise exceptions.InvalidArgumentCombination("Use cache properties OR client_request_token propertie")
+
     if cache_info.has_valid_cache is True:
         _logger.debug("Valid cache found. Retrieving...")
```

Comment:
We already check for the first part of the condition so it can be moved within like so:
```
    if cache_info.has_valid_cache is True:
        _logger.debug("Valid cache found. Retrieving...")
        if client_request_token:
            raise exceptions.InvalidArgumentCombination("One of cache properties OR client_request_token can be used, not both.")
```